### DISCUSSION TITLE
Playlist browser ignores "disc number" when sorting - how to fix?

### DISCUSSION THREAD

**Discussion by amlana4** [DeaDBeeF-Player/deadbeef#3021] [Q&A]
Hi all,

I'm new to DeadBeeF and first of all would like to thank the great work you guys are doing in making this app available for free.

The issue I'm having is that the playlist browser by default seems to sort tracks by "track number" but this ignores the disc number in compilations. Let's say, using a convention "disc number - track number", it is sorting as 1-01, 2-01, 1-02, 1-02 when it should be 1-01, 1-02, 2-01, 2-02.

Please see the problem shown in the screenshot below.

<img width="1334" alt="image" src="https://github.com/DeaDBeeF-Player/deadbeef/assets/152958350/4fea6414-8bb7-45a8-b4cf-af3fd1fa5c72">

I have a large library carefully organised into folders and with the correct file names such that when sorted by path this problem should be rectified - I couldn't get the default to sort by file path though.

I searched this forum but couldn't find an answer to this. Hope you guys can point me to the right direction.

Cheers :)

**Comment by Oleksiy-Yakovenko** [comment]
There's no default sort by the criteria which you need, but you can use custom sort and specify file path titleformat: `%path%`

**Reply by amlana4** [reply]
Thanks for your reply. How do I change the default sorting to %path%? 
I tried Edit > Sort current playlist by > Custom, but this only works temporarily for that playlist. When I click on other folders, it resets to the default. Couldn't find where to change the global settings.

**Comment by Oleksiy-Yakovenko** [comment]
I think you should be able to use the Autosort feature, which is accessible from playlist context menu:
<img width="528" alt="Screenshot 2023-12-13 at 17 24 49" src="https://github.com/DeaDBeeF-Player/deadbeef/assets/1895698/0da84526-6238-47ac-8fc3-09e80a919c0c">

**Reply by amlana4** [reply]
Is this option available on Mac? I don't see those playlist tabs in the Mac version, neither could find a way to enable them.
See below how my player looks like.

<img width="961" alt="image" src="https://github.com/DeaDBeeF-Player/deadbeef/assets/152958350/dfde0867-4e3e-45fe-9159-fce09d6bef23">

**Reply by Oleksiy-Yakovenko** [reply]
Yes, you need to replace the playlist widget in design mode, with the one called "Playlist with tabs"

**Reply by amlana4** [reply]
Thanks! I did that, but unfortunately the Autosort does not remember the custom sorting - it resets every time to the default.

Would it be possible to implement sorting respecting disc number? That is needed in order to display compilations correctly. Compilations are very common, so I believe this feature would benefit many other users too.

**Comment by fabertawe** [comment]
My albums sort correctly by Disc Number. These are my settings...

Playlist (tab) -> Enable Autosort -> yes

These are my columns within the playlist (tab)...
`Album Art`  %artist%%year%%album%%discnumber%%tracknumber%
`♫`
`Artist` %album artist%%year%%album%%discnumber%%tracknumber%
`#` %discnumber%%tracknumber%
`Title` %track%
`Duration`
`Year`

Group By - not used

Hope that helps.

**Reply by amlana4** [reply]
Do you mean you have a column with "title format" = %album art% and "sort format" = %artist%%year%%album%%discnumber%%tracknumber% etc?

If that is the correct interpretation, I tried your settings and they didn't work. Actually, whenever I disable "Group By" then no artworks are displayed. Are you also using the Mac version?

**Reply by fabertawe** [reply]
"Add Column" or "Edit Column" -> Title "Album Art" -> Type "Album Art" -> Sort format "%artist%%year%%album%%discnumber%%tracknumber%"

And so on, exactly as I posted previously :)

I'm not on Mac, I'm using Linux (Arch).

**Reply by fabertawe** [reply]
I didn't mean I had "Group by -> none". I think I must have used "Artist/Date/Album", on checking it now. There is no selection marker to show what I did use, so I assumed nothing was used! If I add any new songs I do (re)sort by clicking the "Album Art" column header.
